clippy_utils/src/lib.rs

```diff
@@ -2791,16 +2791,46 @@ impl<'tcx> ExprUseCtxt<'tcx> {
                         .position(|arg| arg.hir_id == self.child_id)
                         .map_or(0, |i| i + 1),
                 ),
-                ExprKind::Field(_, name) => ExprUseNode::FieldAccess(name),
+                ExprKind::Field(_, name) => ExprUseNode::FieldAccess(use_expr.hir_id, name),
                 ExprKind::AddrOf(kind, mutbl, _) => ExprUseNode::AddrOf(kind, mutbl),
+                ExprKind::Assign(lhs, rhs, _) => {
+                    debug_assert!(lhs.hir_id == self.child_id || rhs.hir_id == self.child_id);
+                    #[allow(clippy::bool_to_int_with_if)]
```

Comment:
NIT:
```suggestion
                    #[expect(clippy::bool_to_int_with_if)]
```
Is there a reason why you rather expect this lint over taking the suggestion?